File: Two Pointers/Easy/LET_no1099_Two Sum Less Than K.py
class Solution:
    def twoSumLessThanK(self, nums: List[int], k: int) -> int:
        nums.sort()
        left, right = 0, len(nums) - 1
        res = float('-inf')
        
        while left < right:
            cur = nums[left] + nums[right]
            if cur < k:
                res = max(res, cur)
                left += 1
            else:
                right -= 1
        
        return res if res != float('-inf') else -1

File: Two Pointers/Easy/LET_no1099_Two Sum Less Than K.cpp
class Solution {
public:
    int twoSumLessThanK(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int res = INT_MIN;
        int left = 0, right = nums.size() - 1;
        
        while (left < right) {
            int cur = nums[left] + nums[right];
            if (cur < k) {
                res = max(res, cur);
                left++;
            }
            else
                right--;
        }
        
        return res > INT_MIN ? res : -1;
    }
};